### PR TITLE
Add `DeprecatedComponents` ERB lint rule

### PR DESCRIPTION
**Pull request by khiga8**
Related: https://github.com/github/accessibility/issues/1290

## Context

We previously added a rubocop rule against deprecated Primer components (https://github.com/primer/view_components/pull/1108) to flag when consumers are using deprecated components.

However, we haven't been able to turn this on for ERB files in dotcom due to limitations of disabling rubocop in ERB files which sucks because there's nothing that's stopping a consumer from using a deprecated primer component.

This PR adds an ERB lint counter equivalent for ERB.

## Updates

- added ERB rule
- pulled methods shared between Rubocop into a shared helpers file
- added tests

### CODE REVIEW

File: lib/primer/view_components/linters/deprecated_components_counter.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          else
+            # add comment with counter if none
+            corrector.insert_before(processed_source.source_buffer.source_range, "#{offense.context}\n")
+          end
+        end
+      end
+
+      private
+
+      def counter_correct?(processed_source)
```
Comment (khiga8) [inline lib/primer/view_components/linters/deprecated_components_counter.rb:51]: I saw that `BaseLinter` already defines this. However, `BaseLinter` seems to be tied to a very particular linter implementation, and has a bunch of things that are irrelevant for the check we're trying to do here. Therefore, I am NOT using `BaseLinter` and re-defining the rule. 

I don't understand the `counter_correct?` in `BaseLinter` enough to know whether its refactorable.

**Comment by camertron** [comment]
@khiga8 I'm not sure I understand 😅 Is this a re-implementation of the ERB counting mechanism available in dotcom?

Side note: I got around the ignore problem by implementing a [config option](https://github.com/primer/view_components/blob/main/lib/primer/view_components/linters/disallow_action_list.rb#L14).

**Comment by khiga8** [comment]
> Is this a re-implementation of the ERB counting mechanism available in dotcom?

yes! there's a `counter_correct?` in `BaseLinter` but it has a bunch of extra things and I don't know what it's doing. 🤔 the `BaseLinter` seems to be specialized but maybe I'm mistaken.

**Comment by khiga8** [comment]
side note, I installed this branch on a dotcom branch!

There are so many violations. here's a draft branch with the changes - [kh-deprecated-components](https://github.com/github/github/compare/kh-deprecated-components-rule?expand=1) (staff only)

File: lib/primer/view_components/linters/deprecated_components_counter.rb (hunk 1 above)
Comment (khiga8) [reply lib/primer/view_components/linters/deprecated_components_counter.rb]: Update: I removed this definition.

Instead I added `require "erblint-github/linters/custom_helpers"` and pull in the method definition there so we aren't duplicating it.

**Review by jonrohan — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**